hypha/apply/projects/permissions.py

```diff
@@ -39,7 +39,27 @@ def can_upload_contract(user, project):
     return False, 'Forbidden Error'
 
 
+def can_access_project(user, project):
+    if not user.is_authenticated:
+        return False, 'Login Required'
+
+    if user.is_contracting and project.status in [CONTRACTING, WAITING_FOR_APPROVAL]:
+        return True, 'Contracting can view project only in Waiting for approval and contracting status'
+
+    if user.is_finance and project.status in [WAITING_FOR_APPROVAL, IN_PROGRESS]:
+        return True, 'Finance can view project only  in Waiting for approval and in progress status'
+
+    if user.is_apply_staff:
+        return True, 'Staff can view project in all statuses'
+
+    if user.is_applicant and user == project.user:
+        return True, 'Applicant(project user) can view project in all statuses'
+
+    return False, 'Forbidden Error'
+
+
 permissions_map = {
     'contract_approve': can_approve_contract,
     'contract_upload': can_upload_contract,
+    'project_access': can_access_project,
```

Comment:
Also, it would be better to have these strings in `CAPITAL_CASE` as these are more like constants.

Reply:
I was also thinking to use `project_view` but that ProjectDetailView contains multiple forms(actions/buttons) as well so it is like 'view project + access to those actions'. And if we want to have those forms' view permission separately then it will be just many permission but yes at the same time, it might give us more flexibility over every form/button.

These permissions are not confirmed yet so let's wait for the confirmation, we can work on it after that.

Reply:
If you looking to add permissions for the lower-level permissions, and they are configurable, I would like to have separate permissions for them.